test/fixtures/mock_server.js

```diff
@@ -0,0 +1,77 @@
+const http = require('http');
+
+export default class MockServer {
```

Comment:
ocd nit: can we name this something like `DummyServer`? technically it's not mocking, it *is* a server.